### PR TITLE
add tranpose, tobool, tonumber, tostring

### PR DESCRIPTION
**Pull request by aq17**

### CODE REVIEW

File: std/tonumber.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
+
+type Tonumber struct{}
+type TonumberArgs struct {
+	Input interface{} `pulumi:"input"`
+}
+
+type TonumberResult struct {
+	Result interface{} `pulumi:"result"`
```
Comment (Zaid-Ajaj) [inline std/tonumber.go:31]: Shouldn't result here be `float64`?

File: std/tostring.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/pulumi/pulumi-go-provider/infer"
+)
+
+type Tostring struct{}
+type TostringArgs struct {
+	Input interface{} `pulumi:"input"`
+}
+
+type TostringResult struct {
+	Result interface{} `pulumi:"result"`
```
Comment (Zaid-Ajaj) [inline std/tostring.go:30]: Same here, should probably be `string`

**Review by Zaid-Ajaj — CHANGES_REQUESTED** [review]
Looks good, just tiny changes regarding tostring/tnumber functions

File: std/tonumber.go (hunk 1 above)
Comment (aq17) [reply std/tonumber.go:31]: According to tf docs, `null` will return `null` , wasn't sure if we should pass `interface{}` vs `*float64`?

File: std/tostring.go (hunk 2 above)
Comment (aq17) [reply std/tostring.go:30]: Same as above ^

Comment (aq17) [reply std/tostring.go:30]: bump 😅

**Comment by Zaid-Ajaj** [comment]
@aq17 nullable numbers as `*float64` and nullable strings as `*string` and I believe we need `pulumi:"result,optional"` to get the type right on the schema side of things. Sorry for the delayed response

**Comment by aq17** [comment]
@Zaid-Ajaj no problem, that sounds good 👍🏼

**Review by Zaid-Ajaj — APPROVED** [review]
Looks great now! only `tobool` I think needs to an update to the output type but so that can be done later 😁

**Pull request merged**